### PR TITLE
Add a `@passmissing` flag, only with `@byrow` for now. 

### PR DESCRIPTION
**Pull request by pdeffebach**

### CODE REVIEW

File: src/macros.jl [hunk 1]
```diff
@@ -283,6 +283,73 @@ macro byrow(args...)
...
 
 
+"""
+    passmissing(args...)
+
+Propograte missing values inside DataFramesMeta.jl macros.
+
+
+`@passmissing` is not a "real" Julia macro but rather serves as a "flag"
+to indicate that the anonymous function created by DataFramesMeta
```
Comment (bkamins) [inline src/macros.jl:293]:
```suggestion
to indicate that the anonymous function created by DataFramesMeta.jl
```

File: src/macros.jl [hunk 2]
```diff
@@ -283,6 +283,73 @@ macro byrow(args...)
...
+to represent an operation should be wrapped in `passmissing` from Missings.jl.
+
+`@passmissing` can only be combined with `@byrow` or the row-wise versions of macros
+such as `@rtransform` and `@rselect`, etc. If any of the values passed
+to the row-wise anonymous function created by DataFramesMeta with `@byrow`, the
```
Comment (bkamins) [inline src/macros.jl:298]:
```suggestion
to the row-wise anonymous function created by DataFramesMeta.jl with `@byrow`, the
```

File: src/macros.jl [hunk 3]
```diff
@@ -283,6 +283,73 @@ macro byrow(args...)
...
+
+Propograte missing values inside DataFramesMeta.jl macros.
+
+
+`@passmissing` is not a "real" Julia macro but rather serves as a "flag"
+to indicate that the anonymous function created by DataFramesMeta
+to represent an operation should be wrapped in `passmissing` from Missings.jl.
+
+`@passmissing` can only be combined with `@byrow` or the row-wise versions of macros
+such as `@rtransform` and `@rselect`, etc. If any of the values passed
```
Comment (bkamins) [inline src/macros.jl:297]:
```suggestion
such as `@rtransform` and `@rselect`, etc. If any of the arguments passed
```

File: src/macros.jl [hunk 4]
```diff
@@ -283,6 +283,73 @@ macro byrow(args...)
...
+to the row-wise anonymous function created by DataFramesMeta with `@byrow`, the
+result will automatically be `missing`.
+
+In the below example, `@transform` would throw an error without the `@passmissing`
+flag.
+
+`@passmissing` is especially useful for functions which operate on `String`s, such as
```
Comment (bkamins) [inline src/macros.jl:304]:
```suggestion
`@passmissing` is especially useful for functions which operate on strings, such as
```

**Comment by bkamins** [comment]
Also index.md needs an update I think (as it currently mentions `passmissing`). Where is the implementation? I understand you will add it later?

**Review by pdeffebach — COMMENTED** [review]
Thanks! @bkamins I just pushed the index.md stuff as well. 

So all the moving parts are done and if it looks good it can be merged.

File: docs/src/index.md [hunk 5]
```diff
@@ -21,6 +21,7 @@ In addition, DataFramesMeta provides
 * `@eachrow` and `@eachrow!` for looping through rows in data frame, again with high performance and 
   convenient syntax. 
 * `@byrow` for applying functions to each row of a data frame (only supported inside other macros).
+* `@passmissing` for propagating missing values inside DataFramesMeta.jl transformations.
```
Comment (bkamins) [inline docs/src/index.md:24]: in line 6: DataFramesMeta.jl (I cannot correct it in the editor)

Comment (bkamins) [inline docs/src/index.md:24]: row-wise transformations?

File: docs/src/index.md [hunk 6]
```diff
@@ -333,6 +334,72 @@ used, functions do not take into account the grouping, so for
...
 
+## Propagating missing values with `@passmissing`
+
+Many Julia functions to not automatically propagate missing values. For instance, 
+`parse(Int, missing)` will error. 
+
+Missings.jl provides the `passmissing` function-wrapper to help get around these
+roadblocks: `passmissing(f)(args...)` will return `missing` if any of `args` is
+missing. Similarly, DataFramesMeta.jl provides the `@passmissing` function to wrap
+the anonymous functions created by DataFramesMeta.jl in `Missings.passmissing`.
```
Comment (bkamins) [inline docs/src/index.md:345]: row-wise

File: src/parsing.jl [hunk 7]
```diff
@@ -92,6 +96,25 @@ function extract_macro_flags(ex::Expr, exprflags = (;Symbol("@byrow") => Ref(fal
...
+Check that the macro flags are consistent with
+one another. For now this only checks that
+`@passmissing` is only called when `@byrow` is
+also called. In the future we may expand
+this function or eliminate it all together.
+"""
+function check_macro_flags_consistency(exprflags)
+    if exprflags[PASSMISSING_SYM][]
+        if !exprflags[BYROW_SYM][]
+            s = "The `@passmissing` flag is currently only allowed with the `@byrow` flag"
```
Comment (bkamins) [inline src/parsing.jl:112]: do you test against this case?

File: src/parsing.jl [hunk 8]
```diff
@@ -179,8 +202,12 @@ function get_source_fun(function_expr; wrap_byrow::Bool=false)
         end
     end
 
-    if wrap_byrow
-        fun = :(ByRow($fun))
+    if exprflags[BYROW_SYM][]
+        if exprflags[PASSMISSING_SYM][]
+            fun = :(ByRow(Missings.passmissing($fun)))
```
Comment (bkamins) [inline src/parsing.jl:207]: looks good performance wise :).

**Review by bkamins — APPROVED** [review]
Looks good. I approve, but after fixing my minor comments you can ping me and I can have a look again.

File: src/parsing.jl (hunk 7 above)
Comment (pdeffebach) [reply src/parsing.jl:112]: Added.

**Comment by pdeffebach** [comment]
Thanks!

**Pull request merged**